src/submission/validation-clinical/donor.ts

```diff
@@ -138,81 +145,126 @@ const crossFileValidator = async (
-        0;
-
-      const invalidTreatmentIntervals = treatments.filter(treatment => {
-        const treatmentRecord = treatment.clinicalInfo;
-
-        const { treatment_start_interval, treatment_duration } = treatmentRecord;
-
-        const treatmentStartInterval =
-          typeof treatment_start_interval === 'number' ? treatment_start_interval : 0;
-
-        const treatmentDuration = typeof treatment_duration === 'number' ? treatment_duration : 0;
-
-        const totalTreatmentTime = treatmentStartInterval + treatmentDuration;
-
-        return totalTreatmentTime > lostToFollowUpInterval;
-      });
-
-      const invalidFollowUpIntervals = followUps.filter(followUp => {
-        const followUpRecord = followUp.clinicalInfo;
-
-        const { interval_of_followup } = followUpRecord;
-
-        const intervalOfFollowUp =
-          typeof interval_of_followup === 'number' ? interval_of_followup : 0;
-
-        return intervalOfFollowUp > lostToFollowUpInterval;
-      });
-
-      const invalidRecords = [...invalidTreatmentIntervals, ...invalidFollowUpIntervals];
-
-      if (invalidRecords.length) {
-        const firstInvalidTreatmentMatch = invalidRecords[0].clinicalInfo;
-        const invalidTreatmentDiagnosisId =
-          firstInvalidTreatmentMatch?.submitter_primary_diagnosis_id;
+      const lostToFollowUpClinicalInfo = donorClinicalEventIdMatch.clinicalInfo;
+
+      const lostToFollowUpStartInterval =
+        Number(
+          lostToFollowUpClinicalInfo.interval_of_followup ||
+            lostToFollowUpClinicalInfo.treatment_start_interval ||
+            lostToFollowUpClinicalInfo.specimen_acquisition_interval,
+        ) || 0;
+
+      const lostToFollowUpDuration =
+        Number(
+          lostToFollowUpClinicalInfo.treatment_duration ||
+            lostToFollowUpClinicalInfo.specimen_duration,
+        ) || 0;
+
+      const lostToFollowUpInterval = lostToFollowUpStartInterval + lostToFollowUpDuration;
+
+      // Collects all Entity Records w/ Treatment Intervals after Lost to Follow Up
+      const invalidClinicalIntervalRecords = [...treatments, ...specimens]
+        .filter(entityRecord => {
+          const { clinicalInfo } = entityRecord;
+          const { treatment_start_interval, treatment_duration } = clinicalInfo;
+          const { specimen_acquisition_interval, specimen_duration } = clinicalInfo;
+
+          const treatmentInterval =
+            typeof treatment_start_interval === 'number'
+              ? treatment_start_interval + (Number(treatment_duration) || 0)
+              : 0;
+
+          const specimenInterval =
+            typeof specimen_acquisition_interval === 'number'
+              ? specimen_acquisition_interval + (Number(specimen_duration) || 0)
+              : 0;
+
+          return (
+            treatmentInterval > lostToFollowUpInterval || specimenInterval > lostToFollowUpInterval
+          );
+        })
```

Comment:
I was surprised that TS was fine letting you deconstruct these properties from objects of two different types... but of course we don't know the structure of ClinicalInfo objects, they are basically just JSON objects, so this works...

In any case, I see what you are doing here but its a bit confusing handling entities of two different types the same way. Would you consider writing some helper methods to clean this a bit? 

`getTreatmentInterval(treatment: DeepReadonly<Treatment>): number` and `getSpecimenInterval(DeepReadonly<Specimen>): number`

and from that:

```ts
const invalidClinicalIntervalRecords = [
    ...treatments.filter(treatment => getTreatmentInterval(treatment) > lostToFollowUpInterval),
    ...specimens.filter(specimen => getSpecimenInterval(treatment) > lostToFollowUpInterval)
].map(record => record.clinicalInfo);
```

Reply:
Yes, agreed. Now that the logic is all sorted I can straighten this out.

Reply:
Pushed an update